pkg/controller/agentregistration/klusterletaddonconfig_controller_test.go

```diff
@@ -0,0 +1,148 @@
+package agentregistration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/library-go/pkg/operator/events/eventstesting"
+	klusterletaddonconfigv1schema "github.com/stolostron/klusterlet-addon-controller/pkg/apis"
+	klusterletaddonconfigv1 "github.com/stolostron/klusterlet-addon-controller/pkg/apis/agent/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestReconcile(t *testing.T) {
+	// Create a new KlusterletAddonConfig object
+	kac := &klusterletaddonconfigv1.KlusterletAddonConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-managed-cluster-with-label",
+			Namespace: "test-managed-cluster-with-label",
+		},
+		Spec: klusterletaddonconfigv1.KlusterletAddonConfigSpec{
+			ApplicationManagerConfig: klusterletaddonconfigv1.KlusterletAddonAgentConfigSpec{
+				Enabled: true,
+			},
+			CertPolicyControllerConfig: klusterletaddonconfigv1.KlusterletAddonAgentConfigSpec{
+				Enabled: true,
+			},
+			IAMPolicyControllerConfig: klusterletaddonconfigv1.KlusterletAddonAgentConfigSpec{
+				Enabled: true,
+			},
+			PolicyController: klusterletaddonconfigv1.KlusterletAddonAgentConfigSpec{
+				Enabled: true,
+			},
+			SearchCollectorConfig: klusterletaddonconfigv1.KlusterletAddonAgentConfigSpec{
+				Enabled: false,
```

Comment:
Thanks for @zhiweiyin318  's explanation.

If you import a cluster from the console, the klustereletaddonconfig CR that the console generated set this field based on whether `search` component is enabled on the hub.

But in our case, util now, we set this to `false` for simplicity.